sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -327,12 +328,17 @@ RT::PiProgram ProgramManager::createPIProgram(const RTDeviceBinaryImage &Img,
         "SPIR-V online compilation is not supported in this context",
         PI_INVALID_OPERATION);
 
+  assert(((Format == PI_DEVICE_BINARY_TYPE_SPIRV) ||
+          (Format != PI_DEVICE_BINARY_TYPE_SPIRV && Devices.size() == 1)) &&
+         "Creating a program from AOT binary for multiple device is not "
+         "supported");
+
   // Load the image
   const ContextImplPtr Ctx = getSyclObjImpl(Context);
   RT::PiProgram Res =
       Format == PI_DEVICE_BINARY_TYPE_SPIRV
           ? createSpirvProgram(Ctx, RawImg.BinaryStart, ImgSize)
-          : createBinaryProgram(Ctx, Device, RawImg.BinaryStart, ImgSize);
+          : createBinaryProgram(Ctx, Devices[0], RawImg.BinaryStart, ImgSize);
```

Comment:
It seems no devices beyond index 0 are ever accessed. Why changing function signature then? If this is for some future extension - please add a comment then.

Reply:
They are accessed when creating from SPIRV. I hope assert clarifies it a bit.

Reply:
Hm. `Devices` parameter is used 3 times in this function, as far as I can see: 
```cpp
for (const device &Dev : Devices) { // debug print Dev }
assert((Format == PI_DEVICE_BINARY_TYPE_SPIRV || Devices.size() == 1)
createBinaryProgram(Ctx, Devices[0], RawImg.BinaryStart, ImgSize);
```
So I don't see how any other device is accessed. Am I missing somthing?

Reply:
Reverted this change.